vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -229,16 +229,20 @@ def test_validate_step_invalid(self):
         "marks, expected",
         [
             ({i: str(i) for i in range(0, 10, 5)}, {i: str(i) for i in range(0, 10, 5)}),
-            ({15: 15, 25: 25}, {15.0: "15", 25.0: "25"}),
-            ({"15": 15, "25": 25}, {15.0: "15", 25.0: "25"}),
+            ({15: 15, 25: 25}, {15: "15", 25: "25"}),  # all int
+            ({15.5: 15.5, 25.5: 25.5}, {15.5: "15.5", 25.5: "25.5"}),  # all floats
+            ({15.0: 15, 25.0: 25}, {15: "15", 25: "25"}),  # all floats, but convertible to int
+            ({"15": 15, "25": 25}, {15: "15", 25: "25"}),  # all string
             (None, None),
         ],
     )
     def test_valid_marks(self, marks, expected):
         range_slider = vm.RangeSlider(min=0, max=10, marks=marks)
-
         assert range_slider.marks == expected
 
+        if marks:
+            assert all(type(key) is type(next(iter(expected.keys()))) for key in range_slider.marks)
```

Comment:
```suggestion
       assert [type(result_key) for result_key in range_slider.marks] == [type(expected_key) for expected_key in expected]
```

Reply:
Thanks!! I like the first 👍 It prints out the types as well.